src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -9,7 +9,8 @@
 
   <Target Name="FindTestDirectories">
     <ItemGroup>
-      <AllTestAssemblies Include="$(BaseOutputPathWithConfig)*Tests\*Tests.dll" />
+      <AllTestAssemblies Condition="'$(SingleTestAssembly)' == ''" Include="$(BaseOutputPathWithConfig)*Tests\*Tests.dll" />
+	  <AllTestAssemblies Condition="'$(SingleTestAssembly)' != ''" Include="$(BaseOutputPathWithConfig)*$(SingleTestAssembly).Tests\*Tests.dll" />
```

Comment:
I'm a little concerned that we are making too many assumptions about the names of assemblies (if we weren't already). We already assume a lot about what test assemblies are named (which isn't actually always true for some of our tests), and this here seems to add the assumption that test assembly names have the product assembly name somewhere in them. In general I think this seems a bit too fragile. What do others think?